tests/swig.at

```diff
@@ -57,6 +58,44 @@ m4_define([POSITIVE_SWIG_TEST],[
   ])
 ])
 
+dnl Positive testcase for invoking souffle with SWIG option given language and test name
+dnl $1 test name
+dnl $2 category
+dnl $3 language
+m4_define([TEST_EVAL_SWIG_WITH_STDOUT],[
+  m4_define([TESTNAME],[$1])
+  m4_define([CATEGORY],[$2])
+  m4_define([LANGUAGE],[$3])
+  m4_define([TESTDIR],["$TESTS"/CATEGORY/LANGUAGE/TESTNAME])
+  m4_define([FACTS],[TESTDIR/facts])
+  m4_define([CURRENTDIR],[`pwd`])
+
+  for file in `ls TESTDIR | grep -v .out$`
+  do
+    cp -R "TESTDIR/$file" .
+  done
+
+  AT_CHECK(["$SOUFFLE" -s LANGUAGE TESTNAME.dl 1>TESTNAME.out 2>TESTNAME.err], [0])
+
+  LANGUAGE_TEST([CURRENTDIR],[LANGUAGE],[TESTNAME])
+
+  SAME_FILE([TESTNAME-LANGUAGE.out],[TESTDIR/TESTNAME-LANGUAGE.out])  

```

Comment:
Is the output deterministic? Would this test still make sense if you sorted the outputs first to make it more reliable?

Reply:
fixed